config.py

```
@@ -63,0 +1,272 @@
+        else:
+            logger.info(".env file not found at '%s'. Skipping .env load.", env_path)
+            self._env_file_loaded = False
+
+        # Load YAML config
+        if os.path.exists(config_path):
+            try:
+                with open(config_path, 'r', encoding='utf-8') as f:
+                    self._config = yaml.safe_load(f) or {} # Ensure _config is a dict
+                self._loaded = True # Indicates YAML file was processed
+                logger.info("Loaded YAML configuration from '%s'.", config_path)
+            except Exception as e:
```

Ruff (0.11.9)

Do not catch blind exception: `Exception`

(BLE001)